nibabel/streamlines/tests/test_tractogram.py

```diff
@@ -181,6 +247,46 @@ def test_getitem(self):
             assert_arrays_equal(sdict[-1][k], v[-1])
             assert_arrays_equal(sdict[[0, -1]][k], v[[0, -1]])
 
+    def test_extend(self):
+        sdict = PerArrayDict(len(DATA['tractogram']),
+                             DATA['data_per_streamline'])
+
+        new_data = {'mean_curvature': 2 * np.array(DATA['mean_curvature']),
+                    'mean_torsion': 3 * np.array(DATA['mean_torsion']),
+                    'mean_colors': 4 * np.array(DATA['mean_colors'])}
+        sdict2 = PerArrayDict(len(DATA['tractogram']),
+                              new_data)
+
+        sdict.extend(sdict2)
+        assert_equal(len(sdict), len(sdict2))
+        for k in DATA['tractogram'].data_per_streamline:
+            assert_arrays_equal(sdict[k][:len(DATA['tractogram'])],
+                                DATA['tractogram'].data_per_streamline[k])
+            assert_arrays_equal(sdict[k][len(DATA['tractogram']):],
+                                new_data[k])
+
+        # Extending with an empty PerArrayDicts should change nothing.
+        sdict_orig = copy.deepcopy(sdict)
+        sdict.extend(PerArrayDict())
+        for k in sdict_orig.keys():
+            assert_arrays_equal(sdict[k], sdict_orig[k])
+
+        # Test incompatible PerArrayDicts.
+        # Other dict has more entries.
+        new_data = {'mean_curvature': 2 * np.array(DATA['mean_curvature']),
+                    'mean_torsion': 3 * np.array(DATA['mean_torsion']),
+                    'mean_colors': 4 * np.array(DATA['mean_colors']),
+                    'other': 5 * np.array(DATA['mean_colors'])}
+        sdict2 = PerArrayDict(len(DATA['tractogram']), new_data)
+        assert_raises(ValueError, sdict.extend, sdict2)
+
+        # Other dict has the right number of entries but wrong shape.
```

Comment:
Hum - but isn't the error in fact coming from the fact that `mean_color` != `other`, rather than the shape difference?  Maybe you need two tests here, one for the keys and one for the shapes, where the shapes test has an entry with the same name, but a different shape.